platform_plugin_aspects/utils.py

```diff
@@ -49,67 +49,53 @@ def generate_superset_context(  # pylint: disable=dangerous-default-value
 
     if language:
         for dashboard in dashboards:
-            if not dashboard["allow_translations"]:
+            if not dashboard.get("allow_translations"):
                 continue
             dashboard["slug"] = f"{dashboard['slug']}-{language}"
             dashboard["uuid"] = str(get_uuid5(dashboard["uuid"], language))
 
-    superset_token, dashboards = _generate_guest_token(
-        user=user,
-        course=course,
-        superset_config=superset_config,
-        dashboards=dashboards,
-        filters=filters,
+    superset_url = _fix_service_url(superset_config.get("service_url"))
+
+    guest_token_url = reverse(
+        "platform_plugin_aspects:superset_guest_token",
+        kwargs={"course_id": course},
     )
 
-    if superset_token:
-        superset_url = _fix_service_url(superset_config.get("service_url"))
-        context.update(
-            {
-                "superset_token": superset_token,
-                "superset_dashboards": dashboards,
-                "superset_url": superset_url,
-            }
-        )
-    else:
-        context.update(
-            {
-                "exception": str(dashboards),
-            }
-        )
+    context.update(
+        {
+            "superset_dashboards": dashboards,
+            "superset_url": superset_url,
+            "superset_guest_token_url": guest_token_url,
+        }
+    )
 
     return context
 
 
-def _generate_guest_token(user, course, superset_config, dashboards, filters):
+def generate_guest_token(user, course, dashboards, filters) -> str:
     """
-    Generate a Superset guest token for the user.
+    Generate and return a Superset guest token for the user.
+
+    Raise ImproperlyConfigured if the Superset API client request fails for any reason.
 
     Args:
         user: User object.
-        course: Course object.
+        course: Course object, used to populate `filters` template strings.
+        dashboards: list of dashboard UUIDs to grant access to.
+        filters: list of string filters to apply.
 
     Returns:
         tuple: Superset guest token and dashboard id.
         or None, exception if Superset is missconfigured or cannot generate guest token.
     """
+    superset_config = settings.SUPERSET_CONFIG
```

Comment:
This would prevent the XBlock from setting a custom Superset URLs. Or was that feature removed already from the XBlock?

Reply:
Yep, overriding Superset config has already been removed from the XBlock.